### ISSUE TITLE
Font Awesome Assets not loading

### ISSUE THREAD

**Issue by kriskropd** [yomli/pico-plugins#2]
None of the icons are loading on the contribute page. Must be something with the way that the font-awesome css is merging. Icons appear if I include `<link rel="stylesheet" href="https://maxcdn.bootstrapcdn.com/font-awesome/4.4.0/css/font-awesome.min.css">` from within the Firefox dev/debug panel after page has loaded, so I'm certain it's not a browser or rendering issue.

**Comment by kriskropd** [comment]
Okay, I've taken another look at this issue. So, Font Awesome assets seem to load correctly on the iPhone's Google Chrome browser. Here are the logs from the contribute page load:

> client.ip.address - - [21/Aug/2015:11:00:45 -0400] "-" 400 0 "-" "-"
> client.ip.address - - [21/Aug/2015:11:01:34 -0400] "POST /contribute HTTP/1.1" 200 398 "site.com/contribute" "Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) CriOS/44.0.2403.67 Mobile/11D257 Safari/9537.53"
> client.ip.address - - [21/Aug/2015:11:01:35 -0400] "GET /contribute HTTP/1.1" 200 2760 "site.com/contribute" "Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) CriOS/44.0.2403.67 Mobile/11D257 Safari/9537.53"
> client.ip.address - - [21/Aug/2015:11:01:36 -0400] "GET /plugins/pico_contributors/assets/css/pico_contributors.MERGED.css HTTP/1.1" 304 0 "site.com/contribute" "Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) CriOS/44.0.2403.67 Mobile/11D257 Safari/9537.53"
> client.ip.address - - [21/Aug/2015:11:01:36 -0400] "GET /plugins/pico_contributors/assets/js/pico_contributors.MERGED.js HTTP/1.1" 304 0 "site.com/contribute" "Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) CriOS/44.0.2403.67 Mobile/11D257 Safari/9537.53"
> client.ip.address - - [21/Aug/2015:11:01:36 -0400] "GET /plugins/pico_contributors/assets/fonts/fontawesome-webfont.woff?v=4.3.0 HTTP/1.1" 304 0 "site.com/contribute" "Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) CriOS/44.0.2403.67 Mobile/11D257 Safari/9537.53"
> client.ip.address - - [21/Aug/2015:11:01:36 -0400] "POST /contribute/assets HTTP/1.1" 200 701 "site.com/contribute" "Mozilla/5.0 (iPhone; CPU iPhone OS 7_1_2 like Mac OS X) AppleWebKit/537.51.2 (KHTML, like Gecko) CriOS/44.0.2403.67 Mobile/11D257 Safari/9537.53"

Meanwhile, when trying to load the contributor page from Firefox on Archlinux (where assets do not appear) the logs print:

> client.ip.address - - [21/Aug/2015:11:08:01 -0400] "GET /contribute HTTP/1.1" 200 2760 "-" "Mozilla/5.0 (X11; Linux x86_64; rv:40.0) Gecko/20100101 Firefox/40.0"
> client.ip.address - - [21/Aug/2015:11:08:01 -0400] "POST /contribute/assets HTTP/1.1" 200 701 "http://site.com/contribute" "Mozilla/5.0 (X11; Linux x86_64; rv:40.0) Gecko/20100101 Firefox/40.0"
> client.ip.address - - [21/Aug/2015:11:08:01 -0400] "GET /plugins/pico_contributors/assets/fonts/fontawesome-webfont.woff2?v=4.3.0 HTTP/1.1" 200 56780 "http://host.ip.address/plugins/pico_contributors/assets/css/pico_contributors.MERGED.css" "Mozilla/5.0 (X11; Linux x86_64; rv:40.0) Gecko/20100101 Firefox/40.0"
> client.ip.address - - [21/Aug/2015:11:08:01 -0400] "GET /plugins/pico_contributors/assets/fonts/fontawesome-webfont.woff?v=4.3.0 HTTP/1.1" 200 64928 "http://host.ip.address/plugins/pico_contributors/assets/css/pico_contributors.MERGED.css" "Mozilla/5.0 (X11; Linux x86_64; rv:40.0) Gecko/20100101 Firefox/40.0"
> client.ip.address - - [21/Aug/2015:11:08:02 -0400] "GET /plugins/pico_contributors/assets/fonts/fontawesome-webfont.ttf?v=4.3.0 HTTP/1.1" 200 66375 "http://host.ip.address/plugins/pico_contributors/assets/css/pico_contributors.MERGED.css" "Mozilla/5.0 (X11; Linux x86_64; rv:40.0) Gecko/20100101 Firefox/40.0"
> client.ip.address - - [21/Aug/2015:11:08:07 -0400] "-" 400 0 "-" "-"
> client.ip.address - - [21/Aug/2015:11:08:07 -0400] "-" 400 0 "-" "-"

And also for Firefox on Macintosh where the icons do not appear:

> client.ip.address - - [21/Aug/2015:11:14:34 -0400] "GET /contribute HTTP/1.1" 200 2760 "-" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0"
> client.ip.address - - [21/Aug/2015:11:14:35 -0400] "GET /plugins/pico_contributors/assets/fonts/fontawesome-webfont.woff2?v=4.3.0 HTTP/1.1" 200 56780 "http://host.ip.address/plugins/pico_contributors/assets/css/pico_contributors.MERGED.css" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0"
> client.ip.address - - [21/Aug/2015:11:14:35 -0400] "GET /plugins/pico_contributors/assets/fonts/fontawesome-webfont.woff?v=4.3.0 HTTP/1.1" 200 71508 "http://host.ip.address/plugins/pico_contributors/assets/css/pico_contributors.MERGED.css" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0"
> client.ip.address - - [21/Aug/2015:11:14:35 -0400] "POST /contribute/assets HTTP/1.1" 200 701 "http://site.com/contribute" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0"
> client.ip.address - - [21/Aug/2015:11:14:35 -0400] "GET /plugins/pico_contributors/assets/fonts/fontawesome-webfont.ttf?v=4.3.0 HTTP/1.1" 200 122092 "http://host.ip.address/plugins/pico_contributors/assets/css/pico_contributors.MERGED.css" "Mozilla/5.0 (Macintosh; Intel Mac OS X 10.10; rv:39.0) Gecko/20100101 Firefox/39.0"
> client.ip.address - - [21/Aug/2015:11:14:40 -0400] "-" 400 0 "-" "-"
> client.ip.address - - [21/Aug/2015:11:14:40 -0400] "-" 400 0 "-" "-"

Maybe the .ttf file is causing the problem? I also noticed the desktops are pulling the absolute paths for pico_contributors.MERGED.css, but they are receiving HTTP 200 so I'm certain they are actually finding the assets requested.

**Comment by yomli** [comment]
Yeah, it's well known that Firefox has issues loading Font Awesome (and any webfont hosted on [your server](https://stackoverflow.com/questions/20032426/fontawesome-doesnt-display-in-firefox), [local](https://bugzilla.mozilla.org/show_bug.cgi?id=760436), or [CDN](http://wpvkp.com/font-awesome-doesnt-display-in-firefox-maxcdn/), btw), due to its cross-origin policy.

I'm using Firefox myself, and Pico Contributors works out-of-the-box on standard LAMP hosting (Gandi and Bitnami). Maybe you can add this to your `.htaccess` :

```
<FilesMatch "\.(ttf|otf|eot|woff)$">
  <IfModule mod_headers.c>
    Header set Access-Control-Allow-Origin "*"
  </IfModule>
</FilesMatch>
```

For Nginx :

```
location ~ /\.(eot|otf|ttf|woff)$ {
        add_header Access-Control-Allow-Origin *;
}
```

I'll add a link to the CDN like you did in the next commit.

**Comment by kriskropd** [comment]
Hm, neither of those seemed to have any effect. I tried the nginx suggestion first, with no difference noted, then the first excerpt,  which also didn't seem to have much effect. Of course, I restarted nginx and php5-fpm services after appending either one to .htaccess - I also cleared cache between reloading the page in Firefox. There weren't any new messages added to the log output either (that I've yet noticed).

**edit**
Actually, I've discovered that this issue lies elsewhere in my configuration. When I visit the http://ip.address/contribute page, the icons seem to load fine, but when I go to the page via my testing domain at http://site.com/contribute, the icons are not loaded. This would seem to be because I had left my `$config['base_url']` in config.php as the servers ip address at the time instead of updating it to the site.com domain I was testing with. I wouldn't have expected this to be an issue (since the paths all lead to the same webfont files), but apparently that was the reason. As expected now, updating the `base_url` to the testing domain resolved this issue (if resolved means the issue now only occurs when visiting the ip address of the server instead of the domain that the pico cms site is attached to).

Also, neither of those .htaccess additions turned out to be necessary for the icons to load in Firefox afterward - though the [article](http://red-team-design.com/firefox-doesnt-allow-cross-domain-fonts-by-default/), where I assume you got those excerpts, was a good read.

Thanks for your help and information.
